### ISSUE TITLE
EventPublishingTestExecutionListener is not included in JUnit 4 and TestNG base classes

### ISSUE THREAD

**Issue by sbrannen** [spring-projects/spring-framework#23748]
The `EventPublishingTestExecutionListener` introduced in Spring Framework 5.2 is not registered automatically for the following base test classes.

- `AbstractJUnit4SpringContextTests`
- `AbstractTransactionalJUnit4SpringContextTests`
- `AbstractTestNGSpringContextTests`
- `AbstractTransactionalTestNGSpringContextTests`

**Comment by patrick-peer** [comment]
This sounds like a good idea, unfortunately this breaks the setup for a particular integration test of mine, beyond repair (well, I could not find a way to work around this, anyway).

The scenario goes like this: I have this Jetty/CXF/Spring server process, which is required to be self contained and start from the command line as a plain old Java application. The setup and tear down of the Spring Context is handled manually within that process, of course.
I have an accompanying integration test, which asserts that the startup and shutdown of the server work as intended. I use TestNG and `AbstractTestNGSpringContextTests` for that.
We recently updated Spring, and this particular test now fails, due to the `EventPublishingTestExecutionListener` accessing the Spring Context in `org.springframework.test.context.support.DefaultTestContext.getApplicationContext()` after it was closed by the server process.

So, while this enhancement improves a lot of cases I guess, I suggest it also requires a possibility to modify the list of listeners.
